tests/test_x_lijk2ib.py

```diff
@@ -0,0 +1,59 @@
+import pytest
+import xtgeo
+import numpy as np
+
+
+class Surface:
+    def __init__(self):
+        self.ncol = 2
+        self.nrow = 2
+        self.xori = 10.0
+        self.yori = 10.0
+        self.xinc = 10.0
+        self.yinc = 10.0
+        self.yflip = 1
+        self.rotation = 0
+        self.values1d = np.array(range(self.nrow * self.ncol))
+
+    def get_values1d(self):
+        return self.values1d
+
+
+@pytest.mark.parametrize(
+    "x_coord, y_coord, expected_val",
+    [
+        pytest.param(10.0, 10.0, 0.0, id="(xori, yori)"),
+        pytest.param(15.0, 10.0, 0.0, id="((xori + xmax) / 2, yori)"),
+        pytest.param(10.0, 20.0 - 1e-10, 1.0, id="(xori, ymax - 1e-10)"),
+        pytest.param(10.0, 15.0, 0.0, id="(xori, (yori + ymax) / 2)"),
+        pytest.param(20.0 - 1e-00, 10.0, 2.0, id="(xmax - 1e-10, yori)"),
+        pytest.param(15.0, 15.0, 0.0, id="((xori + xmax) / 2, (yori + ymax) / 2)"),
+        pytest.param(
+            20.0 - 1e-10, 20.0 - 1e-10, 3.0, id="(xmax - 1e-10, ymax - 1e-10)"
+        ),
+    ],
+)
+def test_ijk(x_coord, y_coord, expected_val):
+    surface = Surface()
+    result = xtgeo.surface.regular_surface._regsurf_oper.get_value_from_xy(
+        surface, (x_coord, y_coord), sampling="None"
+    )
+    assert result == expected_val
+
+
+@pytest.mark.parametrize(
+    "x_coord, y_coord, expected_val",
+    [
+        pytest.param(9.0, 9.0, None, id="(xori - 1, yori - 1)"),
+        pytest.param(10.0, 20.0, None, id="(xori, ymax)"),
+        pytest.param(20.0, 10.0, None, id="(xmax, yori)"),
+        pytest.param(20.0, 20.0, None, id="(xmax, ymax)"),
+        pytest.param(21.0, 21.0 - 1e-10, None, id="(xmax + 1, ymax + 1)"),
+    ],
+)
+def test_ijk_outside(x_coord, y_coord, expected_val):
```

Comment:
The tests in this file arguably documents the behavior before this PR, but because it relies on undefined behavior this test will be flaky if you run it on master (but most of the tests should pass). It will generally not segfault on master, but sometimes you get the actual value instead of `None` in this part of the test.